packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -217,124 +180,75 @@ const DataGrid = injectIntl((props) => {
-    setDataGridColumns(displayedColumns.map((column) => initializeColumn(column)));
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [hasSelectableRows]);
+    // We have to search the dom for the table ref because react-intl v2 does not support forwardRef.
 
-  // useEffect for row displayed columns
-  useEffect(() => {
-    setDataGridColumns(displayedColumns.map((column) => initializeColumn(column)));
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [pinnedColumns, overflowColumns]);
-
-  // useEffect to calculate pinned column offsets
-  useEffect(() => {
-    const offsetArray = [];
-    let cumulativeOffset = 0;
-    let lastPinnedColumnIndex;
-
-    // if grid has selectable rows but no pinned columns, then set the offset of the first column to 0
-    if (hasSelectableRows && pinnedColumns.length === 0) {
-      lastPinnedColumnIndex = 0;
-      offsetArray.push(cumulativeOffset);
-      setPinnedColumnOffsets(offsetArray);
-      return;
-    }
-
-    if (pinnedColumns.length > 0) {
-      offsetArray.push(cumulativeOffset);
-
-      lastPinnedColumnIndex = hasSelectableRows ? pinnedColumns.length : pinnedColumns.length - 1;
-
-      // eslint-disable-next-line array-callback-return
-      dataGridColumns.slice(0, lastPinnedColumnIndex).map((pinnedColumn) => {
-        cumulativeOffset += pinnedColumn.width;
-        offsetArray.push(cumulativeOffset);
-      });
-    }
-    setPinnedColumnOffsets(offsetArray);
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [dataGridColumns]);
+    const [table] = gridContainerRef.current.getElementsByTagName('table');
+    grid.current = table;
+  }, [gridContainerRef]);
 
   // -------------------------------------
 
   const handleMoveCellFocus = (fromCell, toCell) => {
     // Obtain coordinate rectangles for grid container, column header, and new cell selection
-    const gridContainerRect = gridContainerRef.current.getBoundingClientRect();
+    const tableContainer = gridContainerRef.current.firstChild;
+    const gridContainerRect = tableContainer.getBoundingClientRect();
     const columnHeaderRect = grid.current.rows[0].cells[toCell.col].getBoundingClientRect();
     const nextCellRect = grid.current.rows[toCell.row].cells[toCell.col].getBoundingClientRect();
 
     // Calculate horizontal scroll offset for right boundary
     if (nextCellRect.right > gridContainerRect.right) {
-      gridContainerRef.current.scrollBy(nextCellRect.right - gridContainerRect.right, 0);
+      tableContainer.scrollBy(nextCellRect.right - gridContainerRect.right, 0);
     } else {
       // Calculate horizontal scroll offset for left boundary
       let scrollOffsetX = 0;
-      if (pinnedColumnOffsets.length > 0) {
-        if (toCell.col > pinnedColumnOffsets.length - 1) {
-          const lastPinnedColumnRect = grid.current.rows[toCell.row].cells[pinnedColumnOffsets.length - 1].getBoundingClientRect();
+      if (pinnedColumns.length > 0 || hasSelectableRows) {
```

Comment:
I would not expect us to have the "|| hasSelectableRows" change.  This condition should not require a change.